homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,116 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    async_add_entities(ZimiLight(device, api) for device in api.lights)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_name = light.name.strip()
```

Comment:
```suggestion
        self._attr_name = None
```

Reply:
Sorry but this approach does not work for this integration.   The zcc helper returns an identifier which I assign to self._attr_unique_id.   If I then set self._attr_name to None then this identifier is used in the UI as the name.

This is not correct, the identifier returned is unique within the context of a single Zimi Cloud Connect, it is not the human readable name that is set by the end-user.   This human readable name is returned as the name attribute from the helper.